Source/Icebreaker/Services/MatchingService.cs

```diff
@@ -185,12 +197,51 @@ private string GetChannelUserObjectId(ChannelAccount account)
             return JObject.FromObject(account).ToObject<TeamsChannelAccount>()?.AadObjectId;
         }
 
+        /// <summary>
+        /// Parse through a list of pairing information.
+        /// </summary>
+        /// <param name="pairHistory">List of pairing information.</param>
+        /// <returns>Returns pairs from the most recent pairing phase. This can be changed to include
+        /// pairings from less recent pairing phases by changing the `latestIteration` threshold.</returns>
+        private Tuple<Dictionary<string, HashSet<string>>, int> ParsePairHistory(IList<PairInfo> pairHistory)
+        {
+            var pastPairs = new Dictionary<string, HashSet<string>>();
+            int latestIteration = 0;
+
+            foreach (var pair in pairHistory)
+            {
+                latestIteration = (latestIteration >= pair.Iteration) ? latestIteration : pair.Iteration;
+            }
+
+            foreach (var pair in pairHistory)
+            {
+                if (pair.Iteration >= latestIteration && pair.User1Id != null)
```

Comment:
For these types of guard statements it can aid clarity to invert the condition and continue. E.G.
```csharp
if (pair.Iteration < latestIteration || pair.User1Id == null) 
{
   continue;
}
```
Explanation of why to [Minimize Nesting](https://en.wikibooks.org/wiki/Computer_Programming/Coding_Style/Minimize_nesting)

Reply:
Would it be
```
if (!(pair.Iteration < latestIteration || pair.User1Id == null))
{
   continue;
}
```